src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -911,16 +917,59 @@ private function createRequestMatcher(ContainerBuilder $container, string $path
             return $this->requestMatchers[$id];
         }
 
-        // only add arguments that are necessary
-        $arguments = [$path, $host, $methods, $ips, $attributes, null, $port];
-        while (\count($arguments) > 0 && !end($arguments)) {
-            array_pop($arguments);
+        $arguments = [];
+        if ($methods) {
+            $container->register($lid = $id.'.'.ContainerBuilder::hash([MethodRequestMatcher::class, $methods]), MethodRequestMatcher::class)
```

Comment:
Should we create an id that does not reference `$id` which already has a hash of all settings ? This would allow sharing the matchers for subparts.
This could be done by using `'.security.request_matcher.'` or `'.security.request_matcher.part.'` as prefix instead of `$id`.